### ISSUE TITLE
Support usb with cherryusb

### ISSUE THREAD

**Issue by sakumisu** [raspberrypi/pico-sdk#1057]
Dear raspberrypi friends,

I have ported rp2040 usb ip with cherryusb, and provide all the device class demos, see [rp2040 device demo](https://github.com/sakumisu/CherryUSB/tree/master/demo/rp2040/cherryusb), and host driver and demo is on going.

But i do not know if i have the chance to pr cherryusb lib in pico-sdk, can someone review and tell me? If no , i will add the custom cmake script in my repo. Thank you!

**Comment by kilograham** [comment]
nice. we might consider adding a wrapper library that uses CherryUSB if a CMake var is set (or via find_package) (if not as a submodule) ... it will be a while before we get a chance to look at it though.

**Comment by sakumisu** [comment]
Thanks a lot.This can be a slow thing to do, just use your free time.

**Comment by kilograham** [comment]
Hey @sakumisu ... we tend to have a lot going on, so sometimes it takes us a while to get back to things! We would like to add CherryUSB support to the Pico SDK in an upcoming release if you still have interest

**Comment by sakumisu** [comment]
> Hey [@sakumisu](https://github.com/sakumisu) ... we tend to have a lot going on, so sometimes it takes us a while to get back to things! We would like to add CherryUSB support to the Pico SDK in an upcoming release if you still have interest

Really? Yeah, i have interest. We will reschedule support for pico sdk, our pico sdk support is too old, and we will consider how to add cmake easily, now way is that we have the cherryusb.cmake and set some cmake variables to use.

**Comment by kilograham** [comment]
Yeah, feel free to share a branch as you go on the CMake front - we had some similar fun with TinyUSB, because both.

- TinyUSB wants to build standalone with SDK, and run its examples
- SDK wants to include TinyUSB, and pico-examples wants to pull in the tests

**Comment by sakumisu** [comment]
I build a simple project, could you have a time to review? About demo position and library position? If you think LGTM, i will pr for them, thank you!

https://github.com/CherryUSB/pico-examples 
https://github.com/[CherryUSB/pico-sdk](https://github.com/CherryUSB/pico-sdk)

**Comment by kilograham** [comment]
Excellent; thank you - i will have a little play offline to get the feel of things

**Comment by matsobdev** [comment]
So now I know, it is `0` rather that `USBCTRL_REGS_BASE` as a `reg_base` like here: https://github.com/CherryUSB/pico-examples/blob/master/usb/cherryusb/device/cdc_msc/main.c#L15. Didn't try but maybe this was holding RISC-V back.

**Comment by sakumisu** [comment]
> So now I know, it is `0` rather that `USBCTRL_REGS_BASE` as a `reg_base` like here: https://github.com/CherryUSB/pico-examples/blob/master/usb/cherryusb/device/cdc_msc/main.c#L15. Didn't try, but maybe this was holding RISC-V back.
> 
> PS. Nope :D Time tu use "debugprobe" finally for the first time :D

Because regbase is fixed in port, so init with rebase is useless.

**Comment by sakumisu** [comment]
https://github.com/raspberrypi/pico-sdk/pull/2212

**Comment by sakumisu** [comment]
https://github.com/raspberrypi/pico-examples/pull/603

**Comment by kilograham** [comment]
move to 2.1.2 as i may add it as "pre-release" there

**Comment by kilograham** [comment]
moving to 2.2.1 as more work is needed on the pico_stdio_usb stuff and picking between TinyUSB and CherryUSB for that
